lib/till.rb

```diff
@@ -41,20 +46,31 @@ def pounds(pence)
         @order_line_totals[item] = (session[item] * price)
       end
     end
-    @order_subtotal = @order_line_totals.values.inject(:+).to_i
+
+    # Calculate current order totals
+    @TAX_RATE = settings.tax_rate
     @volume_discount_rate = settings.volume_discount_rate
     @volume_discount_threshold = settings.volume_discount_threshold
-    @volume_discount_amount = (@order_subtotal * @volume_discount_rate / 100).to_i
-    @TAX_RATE = settings.tax_rate
-    if @order_subtotal >= @volume_discount_threshold
+    @muffin_discount_rate = settings.muffin_discount_rate
+
+    @order_subtotal = @order_line_totals.values.inject(:+).to_i
+
+    @order_muffin_subtotal = @order_line_totals.select{ |key, value| key.include?('Muffin') }.values.inject(:+).to_i
```

Comment:
Unused block argument - `value`. If it's necessary, use `_` or `_value` as an argument name to indicate that it won't be used.<br>Line is too long. [116/80]<br>Space missing to the left of {.